Tests/Tracking/Repository/IdentifyRepositoryTest.swift

```diff
@@ -147,4 +147,176 @@ class IdentifyRepositoryTest: UnitTest {
 
         waitForExpectations()
     }
+
+    // MARK: trackEvent
+
+    func test_trackEvent_expectCallHttpClientWithCorrectParamsNoBody() {
+        httpClientMock.requestClosure = { params, onComplete in
+            onComplete(Result.success(Data()))
+        }
+
+        let givenIdentifier = String.random
+
+        let expect = expectation(description: "Expect to complete")
+        expect.expectedFulfillmentCount = 2
+        integrationRepository
+            .addOrUpdateCustomer(identifier: givenIdentifier, body: EmptyRequestBody(), jsonEncoder: nil) { result in
+                XCTAssertEqual(self.integrationRepository.identifier, givenIdentifier)
+                expect.fulfill()
+            }
+
+        let givenEventName = String.random
+        let expectedBody = "{\"name\":\"\(givenEventName)\"}".data
+
+        httpClientMock.requestClosure = { params, onComplete in
+            guard case .trackCustomerEvent(let actualIdentifier) = params.endpoint else { return XCTFail() }
+
+            XCTAssertEqual(actualIdentifier, givenIdentifier)
+            XCTAssertEqual(params.body!, expectedBody)
+
+            onComplete(Result.success(params.body!))
+        }
+
+        integrationRepository.trackEvent(name: givenEventName, data: EmptyRequestBody(), timestamp: nil,
+                                         jsonEncoder: nil) { _ in
+            expect.fulfill()
+        }
+
+        waitForExpectations()
```

Comment:
The test function currently tests that `expect` is fulfilled which tests that `repo.trackEvent()`'s `onComplete` gets called however it doesn't check that the http client does get called. Adding assertions against `httpClientMock.requestCalledCount` and other properties would take care of that. 

Same comment for a couple of other test functions below added to this class.